### ISSUE TITLE
FNX3-16591 ⁃ [Bug] Full screen videos don't cover the whole screen

### ISSUE THREAD

**Issue by Lululalu** [mozilla-mobile/fenix#2794]
## Steps to reproduce
Go to YouTube and watch some video in full screen 

### Expected behavior
The video should use the whole screen. 

### Actual behavior
![Screenshot](https://user-images.githubusercontent.com/32398857/58288759-dee4b480-7d8a-11e9-9636-479623d2fbd5.jpg)UNITO-UNDERSCORE!20190523-184323!UNITO-UNDERSCORE!Firefox Preview!

There is empty space above the player. 

### Device information
* Android device: Galaxy A5 2017 8.0
* Fenix version: 1.0.1921

**Comment by vesta0** [comment]
This is especially an issue in landscape mode.

**Comment by jonalmeida** [comment]
This is a bit of an awkward fix. We need to remove the margins from the engine layout when entering/exiting fullscreen. 

Something like this for exiting..

```kotlin
(view.swipeRefresh.layoutParams as CoordinatorLayout.LayoutParams).run {
    val toolbarHeight = resources.getDimension(R.dimen.browser_toolbar_height).toInt()
    setMargins(0, 0, 0, toolbarHeight)
}
```

The awkward part is getting this same thing to work for Custom Tabs as well.

**Comment by sv-ohorvath** [comment]
@ekager Still seeing it the first time I open a youtube video on a Samsung Galaxy Tab S3 tablet (9.7” display). Refresh the page, open a different video or just exit and enter fullscreen again, the white area will be removed.
Not reproducing on a Huawei Media Pad (8.0" display), which also has on-screen buttons vs the physical buttons on the Galaxy Tab.

![Screenshot_20190606-110930_Firefox Preview](https://user-images.githubusercontent.com/4037709/59019773-51589880-8851-11e9-82d6-c2ba45f3befb.jpg)

**Comment by ekager** [comment]
@sv-ohorvath are you still seeing this issue? We can move it into the next sprint to take another look if so

**Comment by vesta0** [comment]
@sv-ohorvath please re-open if you are still seeing this issue.

**Comment by AndiAJ** [comment]
Hi, verified as fixed on the latest Fenix build (#11790617 from 28/6) using the following devices:

• Google Pixel 3XL (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7.0)
• Samsung Galaxy J7 (Android 6.0.1)
• Sony Xperia Z3 (Android 5.1.1)
